hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -109,18 +109,34 @@ func NewAzureDeploymentReconciler(
 	}
 }
 
+func (r *AzureDeploymentReconciler) checkForFatalReconciliationError(err error) {
+	var fatal FatalReconciliationError
+	if errors.As(err, &fatal) {
+		r.SetResourceProvisioningState(armclient.FailedProvisioningState)
+		r.SetResourceError(fatal.Message)
```

Comment:
@matthchr do we need to call `Update` after doing this?

Reply:
Yes, there's not an update baked in at the top level in `generic_controller` right now, so we own calling it before we return if we want to.

Reply:
and in order for modifications to be persisted, we must call update*